openedx/tests/xblock_integration/test_recommender.py

```diff
@@ -650,7 +651,8 @@ def attempt_upload_file_and_verify_result(self, test_case, event_name, content=N
         happens or is rejected as expected.
         """
         if 'magic_number' in test_case:
-            f_handler = StringIO(test_case['magic_number'].decode('hex'))
+            magic_number_hex = codecs.decode(test_case['magic_number'], 'hex_codec')
+            f_handler = StringIO("".join(chr(x) for x in bytearray(magic_number_hex)))
```

Comment:
The bytearray you get from the above decode can just be encoded to utf-8 directly. eg. `magic_number_hex.decode('utf-8')`

Remove the for loop and do that instead.

Reply:
@feanil encoding it directly gives an error "UnicodeDecodeError: 'utf8' codec can't decode byte 0x89 in position 0: invalid start byte"

Reply:
In that case it seems like the input is ascii.  You cane `decode('ascii')` instead right?

Reply:
@feanil decode('ascii') produces the same error. Input is hexa encoded characters. The best solution I could find was to use bytearray. The loop is required because StringIO expects a string input.

Reply:
If decode('hex_codec') is producing bytes, then I think you can just tack on a `decode('utf-8')` at the end of the decode bytes and get a string.

Reply:
@feanil I have added a condition here. The StringIO for python 2 works with a byte input but the StringIO for python 3 requires the input to be a string instead of bytes.